### ISSUE TITLE
some questions about the released code

### ISSUE THREAD

**Issue by vicmax** [SunWenJu123/BMKP#1]
Hi,

Thank you for sharing the code with the community. However, I have some questions about the code. It would be appreciated if I can get some help from you.

- The reason why classifier/BN parameters were not trained since the second task?

We can observe that the optimizer received trainable parameters in the network with:
> https://github.com/SunWenJu123/BMKP/blob/58ea1b151b8ab365de67eb4dd2b93dac2080dde4/models/bmkp.py#L159

it indicated that the classifier parameters will be excluded because:

> https://github.com/SunWenJu123/BMKP/blob/58ea1b151b8ab365de67eb4dd2b93dac2080dde4/models/bmkp.py#L498
```
   def get_params(self) -> torch.Tensor:
        params_arr = []
        for k, (m, params) in enumerate(self.named_parameters()):
            if len(params.size()) == 4:
                params_arr.append(params)
        return params_arr
```
- Evaluation with $W_t$ rather than ${A_{t-1}, B}$ for previous tasks $t-1$.
It seems that the current code did the evaluation on the task $t-1$ with the current weights parameters $W_t$, rather than recovering the weights from $A_{t-1}$ and $B$ described in the supplementary materials. Is it a wrong way?

> https://github.com/SunWenJu123/BMKP/blob/58ea1b151b8ab365de67eb4dd2b93dac2080dde4/utils/training.py#L133

If I misunderstood anything, it will be greatly appreciated if you can point out to correct it.

**Comment by vicmax** [comment]
(continued)
- How to set "track_running_stats=True/False" in BatchNormalization (BN) layers.

In ```main.py```, it seems that the situations of setting "track_running_stats" to "True" and "False" both appeared. I am not sure when to set it as "True" and when to "False"...

> https://github.com/SunWenJu123/BMKP/blob/58ea1b151b8ab365de67eb4dd2b93dac2080dde4/main.py#L55

Looking forward to your help!

**Comment by SunWenJu123** [comment]
Hi, thanks for reaching out. : )

+ Why were the classifier/BN parameters not trained since the second task?
We follow GPM to implement this code.

+ Evaluation with rather than for previous tasks.
We calculate $A_{t-1}B$ in the "forward" function of the model. As follows:

```
    def forward(self, x: torch.Tensor, task_id) -> torch.Tensor:
        weights = self.param_weights[task_id]
        self.composition_net(self.net, weights)   # this step calculate $A_{t-1}B$
        self.net.eval()
        x = x.to(self.device)
        with torch.no_grad():
            outputs = self.net(x, task_id)
        return output
```


+ How to set "track_running_stats=True/False" in BatchNormalization (BN) layers.
By tuning, we empirically set it to True for Tiny-ImageNet and False for CIFAR-100. It depends on the dataset.

**Comment by vicmax** [comment]
Thanks! It helps a lot.
